GitCommands/Git/GitModule.cs

```diff
@@ -3182,111 +3182,219 @@ public GitBlame Blame(string filename, string from, Encoding encoding)
-                    if (line.StartsWith("\t"))
-                    {
-                        blameLine.LineText = line.Substring(1) // trim ONLY first tab
-                                                 .Trim('\r'); // trim \r, this is a workaround for a \r\n bug
-                        blameLine.LineText = ReEncodeStringFromLossless(blameLine.LineText, encoding);
-                    }
-                    else if (line.StartsWith("author-mail"))
-                    {
-                        blameHeader.AuthorMail = ReEncodeStringFromLossless(line.Substring("author-mail".Length).Trim());
-                    }
-                    else if (line.StartsWith("author-time"))
-                    {
-                        blameHeader.AuthorTime = DateTimeUtils.ParseUnixTime(line.Substring("author-time".Length).Trim());
-                    }
-                    else if (line.StartsWith("author-tz"))
-                    {
-                        blameHeader.AuthorTimeZone = line.Substring("author-tz".Length).Trim();
-                    }
-                    else if (line.StartsWith("author"))
-                    {
-                        blameHeader = new GitBlameHeader
-                        {
-                            CommitGuid = blameLine.CommitGuid,
-                            Author = ReEncodeStringFromLossless(line.Substring("author".Length).Trim())
-                        };
-                        blame.Headers.Add(blameHeader);
-                    }
-                    else if (line.StartsWith("committer-mail"))
-                    {
-                        blameHeader.CommitterMail = line.Substring("committer-mail".Length).Trim();
-                    }
-                    else if (line.StartsWith("committer-time"))
-                    {
-                        blameHeader.CommitterTime = DateTimeUtils.ParseUnixTime(line.Substring("committer-time".Length).Trim());
-                    }
-                    else if (line.StartsWith("committer-tz"))
-                    {
-                        blameHeader.CommitterTimeZone = line.Substring("committer-tz".Length).Trim();
-                    }
-                    else if (line.StartsWith("committer"))
-                    {
-                        blameHeader.Committer = ReEncodeStringFromLossless(line.Substring("committer".Length).Trim());
-                    }
-                    else if (line.StartsWith("summary"))
+                    var text = ReEncodeStringFromLossless(line.Substring(1), encoding);
+
+                    GitBlameCommit commit;
+                    if (hasCommitHeader)
                     {
-                        blameHeader.Summary = ReEncodeStringFromLossless(line.Substring("summary".Length).Trim());
+                        commit = new GitBlameCommit(objectId, author, authorMail, authorTime, authorTimeZone, committer, committerMail, committerTime, committerTimeZone, summary, filename);
```

Comment:
Perhaps we should have an analyzer for the max length of a line. This seems way to long.

Reply:
Thanks for taking a look, Janusz.

This was proposed in https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/1808 and there's some discussion there.

For now I'll break the line up.